at_client/lib/src/manager/sync_manager.dart

```diff
@@ -110,8 +113,8 @@ class SyncManager {
       // cloud is ahead if server commit id is > last synced commit id in local
       if (serverCommitId > lastSyncedCommitId) {
         // send sync verb to server and sync the changes to local
-        var syncResponse =
-            await _remoteSecondary.sync(lastSyncedCommitId, regex: regex);
+        var syncResponse = await _remoteSecondary.sync(lastSyncedCommitId,
+            regex: regex, syncCallback: _syncLocal);
         if (syncResponse != null && syncResponse != 'data:null') {
           syncResponse = syncResponse.replaceFirst('data:', '');
           var syncResponseJson = jsonDecode(syncResponse);
```

Comment:
Can we prevent creation of "syncResponse". If we use jsonDecode(syncResponse) with For will it be called everytime?

Reply:
Since we passing a function callback, control does not come into the IF condition. Hence removed the code and tested it. It is working fine.